Wox.Infrastructure/StringMatcher.cs

```diff
@@ -164,12 +151,42 @@ public static int ScoreForPinyin(string source, string target)
     public class MatchResult
     {
         public bool Success { get; set; }
-        public int Score { get; set; }
+
+        private int _score;
+        public int Score
+        {
+            get
+            {
+                return _score;
+            }
+            set
+            {
+                _score = ApplySearchPrecisionFilter(value);
+            }
+        }
         
         /// <summary>
-        /// highlight string
+        /// Matched data to highlight.
         /// </summary>
-        public string Value { get; set; }
+        public List<int> MatchData { get; set; }
+
+        public bool IsSearchPrecisionScoreMet()
+        {
+            return IsSearchPrecisionScoreMet(Score);
+        }
+
+        private bool IsSearchPrecisionScoreMet(int score)
+        {
```

Comment:
Moving these precision methods into the MatchResult class creates a need to new up the object before using it. Would it be better if you have to just use StringMatcher.IsSearchPrecisionsScoreMet? this way i can give it any number and i will know if it meets the required precision.

Reply:
Originally, the methods were implemented as extension methods added to the MatchResult class if I see correctly. I was not entirely sure why they were implemented like that, maybe you can help me here with the background? 
I think the instances should behave the same as before, only without using extension methods, which should normally be the preferred way when possible.

Reply:
So I added them as static methods because I don't think StringMatcher should have a state attached, they should just be even if it's considered helper methods. At the time I don't foresee StringMatcher to grow, be extended, or contain different search logic. 

Mocking is also not needed for static methods, so testing them is a lot easier and straightforward.  I am coming from the perspective of do not introduce state unless there is a real need otherwise as the code base grows it becomes harder and harder to reason about the state the object is in. Particularly true for low level classes like these, do not want the plugin to guess or worry about state of these classes.

Extension methods just provides a simpler and prettier way of calling methods, behavior is the same regardless.